src/frontends/android/app/src/main/java/org/strongswan/android/ui/VpnProfileDetailActivity.java

```diff
@@ -312,30 +312,33 @@ else if (!TextUtils.isEmpty(mUserCertAlias))
 	private void updateCertificateSelector()
 	{
 		if (!mCheckAuto.isChecked())
-		{
+        {
 			mSelectCert.setEnabled(true);
 			mSelectCert.setVisibility(View.VISIBLE);
 
 			if (mCertEntry != null)
-			{
+            {
 				((TextView)mSelectCert.findViewById(android.R.id.text1)).setText(mCertEntry.getSubjectPrimary());
 				((TextView)mSelectCert.findViewById(android.R.id.text2)).setText(mCertEntry.getSubjectSecondary());
 			}
-			else if (!TextUtils.isEmpty(mCaCertCN))
-			{
-				((TextView)mSelectCert.findViewById(android.R.id.text1)).setText(mCaCertCN);
-				mCheckAuto.setVisibility(View.INVISIBLE);
-			}
-		}
+			 else
+            {
+                setCaCertificateDescription();
+            }
+        }
 		else
 		{	mSelectCert.setEnabled(false);
 			mSelectCert.setVisibility(View.GONE);
 		}
 	}
 
+    private void setCaCertificateDescription(){
+        ((TextView)mSelectCert.findViewById(android.R.id.text1)).setText(TextUtils.isEmpty(mCaCertCN)?getString(R.string.certificate_absent): mCaCertCN);
```

Comment:
This line is hardly readable. Maybe extract text to some string before and then set. Also check spaces between ? and :